### ISSUE TITLE
kinesis delivery stream cloud module

### ISSUE THREAD

**Issue by dan-osms** [ansible/ansible#30639]
##### ISSUE TYPE

 - Feature Idea

##### COMPONENT NAME

kinesis delivery stream module

##### ANSIBLE VERSION

```
ansible --version
ansible 2.3.0.0
  config file = /etc/ansible/ansible.cfg
  configured module search path = Default w/o overrides
  python version = 2.7.13 (default, Jul 19 2017, 08:25:13) [GCC 4.2.1 Compatible Apple LLVM 8.1.0 (clang-802.0.42)]
```

##### CONFIGURATION

ANSIBLE_SSH_PIPELINING(/etc/ansible/ansible.cfg) = True
COLOR_DEBUG(/etc/ansible/ansible.cfg) = dark gray
DEFAULT_HOST_LIST(/etc/ansible/ansible.cfg) = ['/etc/ansible/local']
RETRY_FILES_ENABLED(/etc/ansible/ansible.cfg) = False
##### OS / ENVIRONMENT

Mac 10.12.16 Sierra
##### SUMMARY

Not able to build delivery streams
##### STEPS TO REPRODUCE

```yaml
None yet
```
Basically a capability where ansible matches the capabilities of the awscli in scope and form.
 create-delivery-stream
--delivery-stream-name <value>
[--delivery-stream-type <value>]
[--kinesis-stream-source-configuration <value>]
[--s3-destination-configuration <value>]
[--extended-s3-destination-configuration <value>]
[--redshift-destination-configuration <value>]
[--elasticsearch-destination-configuration <value>]

**Comment by clarkst** [comment]
I'd really like to see this module added too.

shdobxr are you working on adding it ?

**Comment by dan-osms** [comment]
To be completely honest I haven't taken a swipe at building modules for ansible. I am no Dev guy by trade, but I get by. 

I am a security arch guy, assisting my company in building out secure infra they can use. My work around is basically the following. (Disclaimer, I will not be held liable for someone running this and getting a bill. Suing me etc. you have been warned ) Sorry had to put this out there.

So I leverage the command module, after generating a skeleton layout. Vastly different than other plays where a module doesn't exist. Also AWS documentation of their template is lacking (IMHO) I was thinking about building a very simplistic json input webpage with a menu driven system to assist the automator in building the template.

```
aws firehose create-delivery-stream --profile <some profile> --generate-cli-skeleton input > file.json
```

```
aws firehose create-delivery-stream --cli-input-json --profile <some profile> file://file.json
```

Where file.json (could be any file really) is a template with variables within it to help keep things dry.

**Comment by clarkst** [comment]
OK I'll take a look at building a module for firehose.

It'll have the minimum functionality initially (meeting my needs of course Stream -> Firehose -> S3 )

**Comment by dan-osms** [comment]
That would be a good start! I wish I could build it myself.

**Comment by clarkst** [comment]
@shdobxr 
Pull request is out there for the initial version
https://github.com/ansible/ansible/pull/31317

**Comment by dan-osms** [comment]
Good deal will pull it and let you know. If satisfied next steps?

**Comment by ofirshtrull** [comment]
any update on this?
Speciali regarding the redshift-destination-configuration and s3-destination-configuration options

**Comment by sivel** [comment]
!component =lib/ansible/modules/cloud/amazon/

**Comment by sivel** [comment]
Thank you very much for your interest in Ansible. Ansible has migrated much of the content into separate repositories to allow for more rapid, independent development. We are closing this issue/PR because this content has been moved to one or more collection repositories.

lib/ansible/modules/cloud/amazon/ -> https://galaxy.ansible.com/community/aws

For further information, please see:
https://github.com/ansible/ansibullbot/blob/master/docs/collection_migration.md
